src/Models/Services/LocalStorageManager/LocalStorageManager.ts

```diff
@@ -0,0 +1,606 @@
+ * @param options options object to be set in the local storage
+ */
+const setEnvironmentOptionsInLocalStorage = (
+    options: EnvironmentOptionsInLocalStorage
+): void => {
+    localStorage.setItem(
+        EnvironmentOptionsLocalStorageKey,
+        JSON.stringify(options)
+    );
+};
+
+/** Updates the selected ADT instance information in local storage
+ * @param selectedAdtInstance currently selected ADT instance resource to be stored in local storage
+ */
+export const setSelectedAdtInstanceInLocalStorage = (
+    selectedAdtInstance: string | IADTInstance
+) => {
+    const environmentConfiguration: EnvironmentConfigurationInLocalStorage =
+        getEnvironmentConfigurationFromLocalStorage() || {};
+    if (selectedAdtInstance) {
+        const selectedAdtInstanceEnvironmentConfigurationItem = getEnvironmentItemFromResource(
+            selectedAdtInstance,
+            AzureResourceTypes.DigitalTwinInstance
+        );
+        if (
+            typeof selectedAdtInstance === 'string' &&
+            areResourceValuesEqual(
+                environmentConfiguration.selectedAdtInstance?.url,
+                selectedAdtInstanceEnvironmentConfigurationItem.url,
+                AzureResourceDisplayFields.url
+            )
+        ) {
+            environmentConfiguration.selectedAdtInstance = {
+                ...selectedAdtInstanceEnvironmentConfigurationItem,
+                id: environmentConfiguration.selectedAdtInstance.id // to preserve the previously tracked id not to override it with null value only when the urls are same
+            };
+        } else {
+            environmentConfiguration.selectedAdtInstance = selectedAdtInstanceEnvironmentConfigurationItem;
+        }
+    } else if (environmentConfiguration?.selectedAdtInstance) {
+        delete environmentConfiguration.selectedAdtInstance;
+    }
+    setEnvironmentConfigurationInLocalStorage(environmentConfiguration);
+};
+
+/** To update the selected Storage account in local storage
+ * @param selectedStorageAccount the storage account to be stored in the local storage
+ */
+export const setSelectedStorageAccountInLocalStorage = (
+    selectedStorageAccount: string | IAzureStorageAccount
+) => {
+    const environmentConfiguration =
+        getEnvironmentConfigurationFromLocalStorage() || {};
+    if (selectedStorageAccount) {
+        const selectedStorageAccountEnvironmentConfigurationItem = getEnvironmentItemFromResource(
+            selectedStorageAccount,
+            AzureResourceTypes.StorageAccount
+        );
+        if (
+            typeof selectedStorageAccount === 'string' &&
+            areResourceValuesEqual(
+                environmentConfiguration.selectedStorageAccount?.url,
+                selectedStorageAccountEnvironmentConfigurationItem.url,
+                AzureResourceDisplayFields.url
+            )
+        ) {
+            environmentConfiguration.selectedStorageAccount = {
```

Comment:
as I added the comment there; it was to preserve the previously tracked id **not to override it with null** value **only when the urls are same**. The scenario is like this: you have a proper selected storage account in local storage with id from previous session (by interacting with EnvironmentPicker), then you load the page again, since only thing we have initially is the container url from which we extract the storage account url, we call setSelectedStorageAccountInLocalStorage method with a string type parameter, then we should not override the existing id if the urls are same. Hope this is clear.